tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -77,6 +90,27 @@ type AzureProvider struct {
 	instanceType string
 	// workspace to store all the results.
 	resourceTrackerDir string
+	// rdp
+	rdp *nsgRuleWrapper
```

Comment:
While I am fine being specific, do you want to create a slice of `*nsgRuleWrapper`? just to make extension easier because I don't see any validation specific to any protocol like `rdp`, `winrm` etc?